agent/reconciler/host_reconciler.go

```diff
@@ -189,72 +189,40 @@ func (r *HostReconciler) cleank8sdirectories(ctx context.Context) error {
-		k8sVersion := byoHost.GetAnnotations()[infrastructurev1beta1.K8sVersionAnnotation]
-		byohBundleTag := byoHost.GetAnnotations()[infrastructurev1beta1.BundleLookupTagAnnotation]
-		bundleInstaller, err := installer.New(bundleRegistry, r.DownloadPath, logger)
+	k8sComponentsInstallationSucceeded := conditions.Get(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded)
+	if k8sComponentsInstallationSucceeded != nil && k8sComponentsInstallationSucceeded.Status == corev1.ConditionTrue {
+		err := r.resetNode(ctx, byoHost)
 		if err != nil {
 			return err
 		}
-		err = bundleInstaller.Uninstall(k8sVersion, byohBundleTag)
-		if err != nil {
-			return err
-		}
-	}
-
-	conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sNodeBootstrapSucceeded, infrastructurev1beta1.K8sNodeAbsentReason, clusterv1.ConditionSeverityInfo, "")
-
-	logger.Info("Removing the bootstrap sentinel file...")
-	if _, err := os.Stat(bootstrapSentinelFile); !os.IsNotExist(err) {
-		err := os.Remove(bootstrapSentinelFile)
-		if err != nil {
-			return errors.Wrapf(err, "failed to delete sentinel file %s", bootstrapSentinelFile)
-		}
-	}
-	if IP, ok := byoHost.Annotations[infrastructurev1beta1.EndPointIPAnnotation]; ok {
-		network, err := vip.NewConfig(IP, registration.LocalHostRegistrar.ByoHostInfo.DefaultNetworkInterfaceName, false)
-		if err == nil {
-			err := network.DeleteIP()
+		if r.SkipInstallation {
+			logger.Info("Skipping uninstallation of k8s components")
+		} else {
+			err = r.uninstallk8sComponents(ctx, byoHost)
 			if err != nil {
 				return err
 			}
 		}
+	} else {
+		logger.Info("Skipping k8s node reset and k8s component uninstallation")
 	}
-	// Remove host reservation
-	byoHost.Status.MachineRef = nil
-
-	// Remove BootstrapSecret
-	byoHost.Spec.BootstrapSecret = nil
-
-	// Remove cluster-name label
-	delete(byoHost.Labels, clusterv1.ClusterLabelName)
-
-	// Remove Byomachine-name label
-	delete(byoHost.Labels, infrastructurev1beta1.AttachedByoMachineLabel)
-
-	// Remove the EndPointIP annotation
-	delete(byoHost.Annotations, infrastructurev1beta1.EndPointIPAnnotation)
-
-	// Remove the cleanup annotation
-	delete(byoHost.Annotations, infrastructurev1beta1.HostCleanupAnnotation)
-
-	// Remove the cluster version annotation
-	delete(byoHost.Annotations, infrastructurev1beta1.K8sVersionAnnotation)
+	conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sNodeBootstrapSucceeded, infrastructurev1beta1.K8sNodeAbsentReason, clusterv1.ConditionSeverityInfo, "")
 
-	// Remove the bundle registry annotation
-	delete(byoHost.Annotations, infrastructurev1beta1.BundleLookupBaseRegistryAnnotation)
+	err := r.removeSentinelFile(ctx, byoHost)
+	if err != nil {
+		return err
+	}
 
-	// Remove the bundle tag annotation
-	delete(byoHost.Annotations, infrastructurev1beta1.BundleLookupTagAnnotation)
+	err = r.deleteEndpointIP(ctx, byoHost)
+	if err != nil {
+		return err
+	}
 
+	err = r.removeAnnotations(ctx, byoHost)
```

Comment:
There is no way `removeAnnotations` is going to return an error. Right? Wondering if we can get rid of the below `if` check for `err`

Reply:
At the moment no, I thought about removing it but kept it in case we add something in future to `removeAnnotations` which returns an error.

Reply:
I think its good to write error handling when we have error being returned. We can remove it now. So that code becomes a little bit more tidy.

Reply:
Cool sounds good